features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListView.kt

```diff
@@ -101,28 +103,52 @@ fun ActionListView(
     val targetItem = (state.target as? ActionListState.Target.Success)?.event
 
     fun onItemActionClicked(
-        itemAction: TimelineItemAction
+        itemAction: TimelineItemAction,
+        immediate: Boolean,
     ) {
         if (targetItem == null) return
-        sheetState.hide(coroutineScope) {
+        if (immediate) {
+            coroutineScope.launch { sheetState.hide() }
             state.eventSink(ActionListEvents.Clear)
             onActionSelected(itemAction, targetItem)
+        } else {
+            sheetState.hide(coroutineScope) {
+                state.eventSink(ActionListEvents.Clear)
+                onActionSelected(itemAction, targetItem)
+            }
```

Comment:
I do not like to modify the code in order to test it, but during the test, the `then` parameter of `SheetState.hide` is not invoked. I will keep thinking about it, and I am open to suggestion to avoid to do this.

Reply:
Are the tests that fail without this code uploaded? I removed it and they seem to be passing just fine locally.

Reply:
Yes, the 2 failing tests were `clicking on send location invoke expected callback` and `clicking on create poll invoke expected callback`.

I will double check again locally, after removing the patch.

Reply:
If I remove the patch, the 2 tests above are still passing, but now `selecting a action on a message emits the expected Event` is failing. Still digging

Reply:
OK, I have fixed the failing test using the test clock: e63e5bd